mws/mws.py

```diff
@@ -117,14 +119,25 @@ class DataWrapper(object):
     """
     Text wrapper in charge of validating the hash sent by Amazon.
     """
-    def __init__(self, data, header):
+    def __init__(self, data, headers):
         self.original = data
         self.response = None
-        if 'content-md5' in header:
+        self.headers = headers 
+        if 'content-md5' in headers:
             hash_ = calc_md5(self.original)
-            if header['content-md5'].encode() != hash_:
+            if headers['content-md5'].encode() != hash_:
                 raise MWSError("Wrong Contentlength, maybe amazon error...")
 
+        if headers['content-type'] == 'application/zip':
+            target_file_name = input("The response is a zip file. Enter name of zip file to store in current folder: ")
```

Comment:
We cannot assume that all users will be using this package on the command line, thus we cannot reliably ask for `input` here.

Instead, we would need to store a file-like object on the response so that end users can save it, read it, etc., as needed.

Reply:
If we put in a UI interface for storing a file-object (windows/mac), is there a way to continue programmatic unzipping and further data manipulations? I'm currently using this wrapper to automatically check if some reports are zip files (and not XML), and if yes, unzip and continue manipulations on the unzipped file (take the output data and feed it into another system). My guess is this functionality would break on a UI.
Any thoughts?

Reply:
Maybe another function definition (eg., get_zipped_file()?)

Reply:
Sorry for the delay in responding :) .

Users typically make use of `python-amazon-mws` as a connector between their other projects and Amazon MWS. For instance, a Django website can get shipment data from their API and store it in data models. Continuing the example, it would not be within the scope of this package to write a Django model field or an admin interface: those would be written in separate projects.

---

We can offer *some* assistance to make things easier for certain general uses. For example, in responses that contain nothing but zipped files, we could transform the response data into a file-like object in memory, which the user might then save to their file system through their own code.

Other responses, like the package label responses in FulfillmentInboundShipments, contain PDFs that are ZIPped and sent as base64-encoded strings (see [here](http://docs.developer.amazonservices.com/en_US/fba_inbound/FBAInbound_ExtractingPdfDocumentData.html)). I don't think we want to assume too much about how a user may want to print these documents, but we could offer a helper method that would parse the base64 string, unzip the resulting ZIP file, extract the PDF contents into a file in-memory, and return that file-like object.